pyanaconda/modules/common/structures/subscription.py

```diff
@@ -362,3 +362,107 @@ def server_proxy_password(self) -> SecretData:
     @server_proxy_password.setter
     def server_proxy_password(self, password: SecretData):
         self._server_proxy_password = password
+
+
+class AttachedSubscription(DBusData):
+    """Data for a single attached subscription."""
+
+    def __init__(self):
+        self._name = ""
+        self._service_level = ""
+        self._sku = ""
+        self._contract = ""
+        self._start_date = ""
+        self._end_date = ""
+        self._consumed_entitlement_count = ""
+
+    @property
+    def name(self) -> Str:
+        """Name of the attached subscription.
+
+        :return: subscription name
+        :rtype: str
+        """
+        return self._name
+
+    @name.setter
+    def name(self, name: Str):
+        self._name = name
+
+    @property
+    def service_level(self) -> Str:
+        """Service level of the attached subscription.
+
+        :return: service level
+        :rtype: str
+        """
+        return self._service_level
+
+    @service_level.setter
+    def service_level(self, service_level: Str):
+        self._service_level = service_level
+
+    @property
+    def sku(self) -> Str:
+        """SKU id of the attached subscription.
+
+        :return: SKU id
+        :rtype: str
+        """
+        return self._sku
+
+    @sku.setter
+    def sku(self, sku: Str):
+        self._sku = sku
+
+    @property
+    def contract(self) -> Str:
+        """Contract identifier.
+
+        :return: contract identifier
+        :rtype: str
+        """
+        return self._contract
+
+    @contract.setter
+    def contract(self, contract: Str):
+        self._contract = contract
+
+    @property
+    def start_date(self) -> Str:
+        """Subscription start date.
```

Comment:
Are we able to specify the format? I know we had some issues with that.

Reply:
The problem is in the JSON from which this is parsed providing dates in an inconsistent manner - for more info see [bug 1793501](https://bugzilla.redhat.com/show_bug.cgi?id=1793501).

The code that does the parsing (coming soon in followup PR) handles these differences and provides a consistent human readable data that is set to the DBus structure and then displayed.

Another issues is that we would like to harmonize the date display between all the tools handling this data (Anaconda, Cockpit RHSM plugin, RHSM CLI, Satellite, etc.) but this is still an ongoing effort and no shared date format has yet been agreed on.

To summarize, the only date format we can specify at the moment for these fields in the DBus structure is "nice human readable date with consistent formatting" - which I will add to the docstring. :)

Later on (possibly before declaring our DBus API stable) we might able able to improve this.

Reply:
Ok, I would document this in the docstring.

Reply:
Added a docstring explaining what can be expected in the data property & linked to the bug.